src/main/java/frc/robot/superstructure/climb/ClimbStateHandler.java

```diff
@@ -0,0 +1,40 @@
+package frc.robot.superstructure.climb;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.subsystems.lifter.LifterState;
+import frc.robot.subsystems.lifter.LifterStateHandler;
+import frc.robot.subsystems.solenoid.SolenoidState;
+import frc.robot.subsystems.solenoid.SolenoidStateHandler;
+
+public class ClimbStateHandler {
+
+	private final LifterStateHandler lifterStateHandler;
+	private final SolenoidStateHandler solenoidStateHandler;
+
+	public ClimbStateHandler(LifterStateHandler lifterStateHandler, SolenoidStateHandler solenoidStateHandler) {
+		this.lifterStateHandler = lifterStateHandler;
+		this.solenoidStateHandler = solenoidStateHandler;
+	}
+
+	public Command setState(ClimbState state) {
+		return switch (state) {
+			case STOP -> lifterStateHandler.setState(LifterState.HOLD).alongWith(solenoidStateHandler.setState(SolenoidState.OFF));
+			case EXTEND ->
+				(lifterStateHandler.setState(LifterState.BACKWARD).raceWith(new WaitCommand(ClimbConstants.SOLENOID_RELEASE_TIME_SECONDS)))
+					.andThen(
```

Comment:
maybe there are more cleanings to this command more than timeouts (i dont have an idea now)

Reply:
well, there is, ill do it without timouts